src/bin/pg_autoctl/pgctl.c

```diff
@@ -1447,6 +1460,61 @@ pg_write_standby_signal(const char *configFilePath,
 }
 
 
+/*
+ * pg_cleanup_standby_mode cleans-up the replication settings for the local
+ * instance of Postgres found at pgdata.
+ *
+ *  - remove either recovery.conf or standby.signal
+ *
+ *  - when using Postgres 12 also make postgresql-auto-failover-standby.conf an
+ *    empty file, so that we can still include it, but it has no effect.
+ */
+bool
+pg_cleanup_standby_mode(uint32_t pg_control_version,
```

Comment:
Is this function still needed now that we don't use `--write-recovery-conf` anymore?

Reply:
We still need to remove `recovery.conf` or `standby.signal` depending on the version of Postgres when promoting a standby to now be a primary. Otherwise at next restart it's not going to be a primary anymore. I don't believe `pg_ctl promote` would do the cleanup for us, and if it did, it would not clean-up the `postgresql-auto-failover-standby.conf` file where we keep the `primary_conninfo` and other replication parameters.